### PR TITLE
Alert on Presence of Zero Logon Vulnerability - Teams

### PR DESCRIPTION
**Pull request by tharward-r7**
TODO:

- [x] Test with a real Teams environment
- [x] test.png
- [ ] Test with VM alert and produce successful job
- [ ] job.png

## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [x] Workflow verified as working in a test
- [x] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [x] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Workflow includes a detailed Description
- [x] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [x] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [x] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [x] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [x] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

----

### Screenshots

workflow1

workflow2

test

teams1

### CODE REVIEW

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+* Insight Platform User API Key
+* [Microsoft Teams](https://insightconnect.help.rapid7.com/docs/configure-slack-for-chatops)
+* InsightVM
+* InsightConnect
+
+# Documentation
+
+## Setup
+
+**In order to use this workflow, you will need to subscribe to the InsightVM Vulnerabilities Found webhook event. You will also need to adjust the Teams team and channel names to match your environment.**
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md:24]:
```suggestion
**In order to use this workflow, you will need to subscribe to the InsightVM Vulnerabilities Found webhook event. You will also need to adjust the Microsoft Teams team and channel names to match your environment.**
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ 
+## Technical Details
+
+Plugins utilized by workflow:
+
+|Plugin|Version|Count|
+|----|----|--------|
+|Rapid7 Vulnerability & Exploit Database|2.0.3|1|
+|Rapid7 InsightVM|4.7.0|1|
+
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md:46]:
```suggestion
|Microsoft Teams|2.2.1|1|
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+name: Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams
+title: Alert on Presence of Zero Logon Vulnerability with Microsoft Teams
+description: Receive Teams messages for any Zero Logon (CVE-2020-1472 or related) vulnerability found
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [vulnerability_management, alerting_and_notifications]
+  keywords: [chat, slack]
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [chat, microsoft_teams]
```

**Review by sengleson-r7 — CHANGES_REQUESTED** [review]

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+status: []
+hub_tags:
+  use_cases: [vulnerability_management, alerting_and_notifications]
+  keywords: [chat, microsoft_teams]
+  features: []
+resources:
+  source_url: https://github.com/rapid7/insightconnect-workflows/tree/master/workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_teams
+  license_url: https://github.com/rapid7/insightconnect-workflows/blob/master/LICENSE
+  screenshots:
+#    - name: slack_output.png
```
Comment (cmchugh-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml:18]: should this be updated to reference the teams screenshot?

**Review by sengleson-r7 — DISMISSED** [review]

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+Once the workflow has been imported, open the workflow and view the InsightVM Events trigger step. Save the trigger configuration step and proceed through the Instructions on subscribing to the InsightVM Vulnerabilities Found Webhook Event. This will require an [Insight Platform User API Key](https://docs.rapid7.com/insight/managing-platform-api-keys#generating-a-user-key). Insert your User API Key, copy the API request, run it from a command prompt, and look for a JSON response including an ID. If you receive a JSON response, then your webhook event subscription is active! Close your command prompt and return to the workflow in InsightConnect.
+
+After subscribing to the vulnerabilities found webhook event, return to the workflow editor and click the "Loop Over found Vulnerabilities" loop. Within the loop, click on the "Post to Microsoft Teams" step, and change the Team and Channel fields from `change_me` to the desired team and channel for the Teams environment where you would like vulnerability notifications delivered. We recommend a private channel with representatives from security and remediation teams for the best visibility.
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md:30]:
```suggestion
After subscribing to the vulnerabilities found webhook event, return to the workflow editor and click the "Loop over Found Vulnerabilities" loop. Within the loop, click on the "Post Message to Teams" step, and change the Team and Channel fields from `change_me` to the desired team and channel for the Teams environment where you would like vulnerability notifications delivered. We recommend a private channel with representatives from security and remediation teams for the best visibility.
```

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+# Version History
+
+* 1.0.0 - Initial workflow
+
+# Links
+
+## References
+
+* [Rapid7 Vulnerability & Exploit Database](https://rapid7.com/db)
+* [Microsoft Teams](https://www.microsoft.com/en-us/microsoft-365/microsoft-teams/group-chat-software)
```
Comment (sengleson-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md:60]:
```suggestion
* [Microsoft Teams](https://www.microsoft.com/en-us/microsoft-365/microsoft-teams/group-chat-software)
* [InsightVM Events Trigger](https://docs.rapid7.com/insightconnect/set-up-an-insightvm-events-trigger)
```

**Review by sengleson-r7 — CHANGES_REQUESTED** [review]

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml (hunk 4 above)
Comment (tharward-r7) [reply workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/workflow.spec.yaml]: Done! Thanks!

**Review by sengleson-r7 — DISMISSED** [review]

File: workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
+# Description
+
+This workflow listens for InsightVM Webhook Events for newly discovered vulnerabilities, checks if a Zero Logon (CVE-2020-1472 or derivitave) vulnerability is present, and if found, posts a notification to a Microsoft Teams channel.
```
Comment (jschipp-r7) [inline workflows/Alert_on_Presence_of_Zero_Logon_Vulnerability_with_Microsoft_Teams/help.md:3]:
```suggestion
This workflow listens for InsightVM Webhook Events for newly discovered vulnerabilities, checks if a Zero Logon (CVE-2020-1472 or derivative) vulnerability is present, and if found, posts a notification to a Microsoft Teams channel.
```

**Review by jschipp-r7 — APPROVED** [review]
@tharward-r7 can merge after retesting

**Review by sengleson-r7 — APPROVED** [review]

**Pull request merged**